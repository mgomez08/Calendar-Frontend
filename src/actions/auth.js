import Swal from "sweetalert2";
import { fetchWithoutToken, fetchWithToken } from "../helpers/fetch"
import { types } from "../types/types";
import { eventLogout } from "./events";


export const startLogin = ( email, password ) => {
    return async( dispatch ) => {
        const resp = await fetchWithoutToken('auth', {email, password}, 'POST');
        const body = await resp.json();

        if( body.ok ){
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            
            dispatch( login({
                uid: body.uid,
                name: body.name,
            }) );
        }else if(body.msg){
            Swal.fire('Error', body.msg, 'error')
        }
    }
}
export const startRegister = ( email, password, name ) => {
    return async( dispatch ) => {
        const resp = await fetchWithoutToken('auth/register', {email, password, name}, 'POST');
        const body = await resp.json();

        if( body.ok ){
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            
            dispatch( login({
                uid: body.uid,
                name: body.name,
            }) );
        }else if(body.msg){
            Swal.fire('Error', body.msg, 'error');
        }
    }
}

export const startChecking = () => {
    return async( dispatch ) => {

        const resp = await fetchWithToken('auth/refreshToken');
        const body = await resp.json();

        if( body.ok ){
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            
            dispatch( login({
                uid: body.uid,
                name: body.name,
            }) );
        }else{
            dispatch( checkingFinish() );
        }
    }
}

const login = ( user ) => ({
    type: types.authLogin,
    payload: user
})

const checkingFinish = () => ({
    type: types.authChekingFinish
})

export const startLogout = () => {
    return (dispatch) => {
        localStorage.clear();
        dispatch(eventLogout());
        dispatch(logout());
    }
}

const logout = () => ({ type: types.authLogout })