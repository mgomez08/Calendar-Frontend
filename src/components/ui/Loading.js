import React from "react";

export const Loading = () => {
  return (
    <>
      <div className="loading">
        <div></div>
        <div></div>
        <div></div>
        <div></div>
        <h1>Cargando informaci√≥n</h1>
      </div>
    </>
  );
};
