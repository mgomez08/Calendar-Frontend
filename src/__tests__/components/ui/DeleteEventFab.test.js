import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import "@testing-library/jest-dom";
import { DeleteEventFab } from "../../../components/ui/DeleteEventFab";
import { eventStartDelete } from "../../../actions/events";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {};
let store = mockStore(initialState);
store.dispatch = jest.fn();

jest.mock("../../../actions/events", () => ({
  eventStartDelete: jest.fn(),
}));

const wrapper = mount(
  <Provider store={store}>
    <DeleteEventFab />
  </Provider>
);

describe("Tests in <DeleteEventFab />", () => {
  test("Test with Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Should call eventStartDelete with click", () => {
    wrapper.find("button").simulate("click");
    expect(eventStartDelete).toHaveBeenCalledWith();
  });
});
