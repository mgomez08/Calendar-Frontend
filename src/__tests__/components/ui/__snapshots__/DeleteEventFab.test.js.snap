// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <DeleteEventFab /> Test with Snapshot 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  >
    <span>
       Delete
    </span>
  </i>
</button>
`;
